projects/kit/components/slider/helpers/slider-key-steps.directive.ts

```diff
@@ -27,45 +62,7 @@ export class TuiSliderKeySteps extends TuiControl<number> {
     );
 
     @Input()
-    public keySteps!: TuiKeySteps;
-
-    public override writeValue(controlValue: number | null): void {
-        if (controlValue === null) {
-            return;
-        }
-
-        const clampedControlValue = tuiClamp(controlValue, this.min, this.max);
-
-        ngDevMode &&
-            console.assert(
-                controlValue === clampedControlValue,
-                '\n[SliderKeySteps]: You cannot programmatically set value which is less/more than min/max',
-            );
-
-        this.slider.value = this.transformToNativeValue(clampedControlValue);
-    }
-
-    protected get min(): number {
-        return this.keySteps[0][1];
-    }
-
-    protected get max(): number {
-        return this.keySteps[this.keySteps.length - 1]?.[1] ?? 0;
-    }
-
-    protected updateControlValue(): void {
-        this.onChange(
-            tuiPercentageToKeyStepValue(this.slider.valueRatio * 100, this.keySteps),
-        );
-    }
-
-    private transformToNativeValue(controlValue: number): number {
-        const {min, max} = this.slider;
-        const newValuePercentage = tuiKeyStepValueToPercentage(
-            controlValue,
-            this.keySteps,
-        );
-
-        return (newValuePercentage * (max - min)) / 100 + min;
+    public set keySteps(steps: TuiKeySteps) {
```

Comment:
We can maybe inject the other directive and use its transformer instead of this setter?

Reply:
It should be overwritten on every input change.
I could use `effect` instead.

Is this solution is better
```ts
private readonly keyStepsBase = inject(TuiSliderKeyStepsBase);
protected readonly $ = effect(
    () => (this.transformer = this.keyStepsBase.transformer()),
);
```

than this one ?
```ts
private readonly slider = inject<TuiSliderComponent>(
    forwardRef(() => TuiSliderComponent),
);

@Input()
public set keySteps(steps: TuiKeySteps) {
    this.transformer = tuiCreateKeyStepsTransformer(steps, this.slider);
}
```